lib/CMakeLists.txt

```diff
@@ -37,15 +37,18 @@ target_link_libraries(lib${PROJECT_NAME} PRIVATE
 # Restricting symbols has several advantages, noted at https://gcc.gnu.org/wiki/Visibility.
 include(GenerateExportHeader)
 generate_export_header(lib${PROJECT_NAME} EXPORT_FILE_NAME "${CMAKE_CURRENT_LIST_DIR}/inc/${PROJECT_NAME}/export.h")
-if ("${CMAKE_SYSTEM_NAME}" MATCHES "Linux")
-    add_compiler_export_flags()
-elseif (WIN32)
+if (NOT APPLE)
     add_compiler_export_flags()
```

Comment:
On AIX it will skip because it's not supported.

On Apple it's supported, but behaves in weird ways. I think mostly it gives warnings about boost, and I don't want to rebuild boost yet.